### ISSUE TITLE
Error upon recompilation on MacOS: Absyn__BREAK

### ISSUE THREAD

**Issue by Mathadon** [OpenModelica/OpenModelica#12476]
### Description

I'm getting the following errors upon recompilation of OM on MacOS:

```
        ~~~~~~~~~~~~~~~~  ^~~~~~~~~
.../build_cmake/OMCompiler/Parser/generated/ModelicaParser.c:11491:29: error: use of undeclared identifier 'Absyn__BREAK'
                             ast = Absyn__BREAK; 
                                   ^
/Users/filipjorissen/Documents/git.nosync/OpenModelica/build_cmake/OMCompiler/Parser/generated/ModelicaParser.c:23308:48: error: call to undeclared function 'Absyn__SUBSCRIPTED_5fEXP'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                                      retval.ast= Absyn__SUBSCRIPTED_5fEXP(el, subs);
                                                  ^

```

It seems that the problematic line of code was introduced through commit `1cf1c15ff35eb3e10e215c51ac76ca01cc02339b` in file `OMCompiler/Parser/Modelica.g`.

### Steps to Reproduce

```
git checkout master
git submodule update --recursive
cmake -S . -B build_cmake -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_Fortran_COMPILER=gfortran -DCMAKE_PREFIX_PATH=/opt/local
cmake --build build_cmake --parallel 10 --target install
```

### Version and OS

Version: current master branch: add1c8e338c4f02eee4b7ff08911130353e80f42
OS: macOS 13.4.1

**Comment by perost** [comment]
It means that the `OpenModelicaBootstrappingHeader.h` header is too old for some reason. You could try deleting the `OMCompiler/Compiler/boot/bomc` folder and see if that fixes the issue.

**Comment by Mathadon** [comment]
@perost thanks for the suggestion. That does not solve the problem. Removing `OMCompiler/Compiler/OpenModelicaBootstrappingHeader.h ` or `build_cmake` did not work either.

A clean reinstall did however solve the problem. 

This is a recurring problem that I'm experiencing when doing incremental updates of OM. I figured that removing `build_cmake` would be sufficient to do a full recompile but apparently this is not the case. Do you know of a more robust way to recompile OM instead of doing a new clone and install?

**Comment by casella** [comment]
Something like `make clean` for CMAKE?

**Comment by mahge** [comment]
You have to do what @perost suggested **and then** reconfigure. When you reconfigure CMake will download the new bootstrapping sources of they are missing. If they are downloaded, then the bootstraped omc will be rebuilt because the files are now. Otherwise, if you do not reconfigure, even if you delete the old sources, nothing will be rebuilt because nothing changed as far as CMake is concerned.

I am not sure why deleting the build folder didn't help. Maybe you tried that only before deleting the bootstrapping sources but not after?

**Comment by Mathadon** [comment]
By 'reconfiguring' you mean running `cmake -S . -B build_cmake`? Perhaps I did not try that in combination with removing with `build_cmake` and `OMCompiler/Compiler/boot/bomc` but I think I did. In any case I can try that when the issue pops up again. I'll not down this issue. Thanks!

**Comment by mahge** [comment]
> By 'reconfiguring' you mean running `cmake -S . -B build_cmake`?

Yes. It's kind of unfortunate how it works now. It can be confusing. At least it does not happen often.

**Comment by bilderbuchi** [comment]
I also [just ](https://github.com/OpenModelica/OpenModelica/issues/11603#issuecomment-2309940296)ran into this (not on MacOS, though). 
It seems to me that such downloaded files should end up (somewhere) in the `CMAKE_BINARY_DIR` (e.g. `cmake_build`), not somewhere in the source tree? Then this problem would not exist, as removing `cmake_build` would clean up as appropriate.

Alternatively, marking those files for cleanup by CMake would also help, I guess: **https://cmake.org/cmake/help/latest/prop_dir/ADDITIONAL_CLEAN_FILES.html**

Luckily, removing the `bomc` directory manually helped in my case.
